react/features/toolbox/components/buttons/VideoQualityButton.web.js

```diff
@@ -0,0 +1,149 @@
+import { connect } from 'react-redux';
+
+import { VideoQualityDialog } from '../../../video-quality';
+
+import ToolbarButton from '../ToolbarButton';
+
+const DEFAULT_BUTTON_CONFIGURATION = {
+    buttonName: 'videoquality',
+
+    // FIXME Temporary icon for use during development.
+    classNames: [ 'button', 'icon-visibility' ],
+    enabled: true,
+    id: 'toolbar_button_videoquality',
+    tooltipKey: 'videoStatus.qualityButtonTip'
+};
+
+const TOOLTIP_TO_DIALOG_POSITION = {
+    bottom: 'bottom center',
+    left: 'left middle',
+    right: 'right middle',
+    top: 'top center'
+};
+
+/**
+ * React {@code Component} for displaying an inline dialog for changing receive
+ * video settings.
+ *
+ * @extends Component
+ */
+class VideoQualityButton extends Component {
+    /**
+     * {@code VideoQualityButton}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Whether or not the button is visible, based on the visibility of the
+         * toolbar. Used to automatically hide the inline dialog if not visible.
+         */
+        _visible: React.PropTypes.bool,
+
+        /**
+         * From which side tooltips should display. Will be re-used for
+         * displaying the inline dialog for video quality adjustment.
+         */
+        tooltipPosition: React.PropTypes.string
+    };
+
+    /**
+     * Initializes a new {@code VideoQualityButton} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showVideoMenu: false
+        };
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onMenuClose = this._onMenuClose.bind(this);
+        this._onMenuOpen = this._onMenuOpen.bind(this);
+    }
+
+    /**
+     * Automatically close the inline dialog if the button will not be visible.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (!nextProps._visible) {
+            this._onMenuClose();
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { _visible, tooltipPosition } = this.props;
+        const buttonConfiguration = {
+            ...DEFAULT_BUTTON_CONFIGURATION,
+            classNames: [
+                ...DEFAULT_BUTTON_CONFIGURATION.classNames,
+                this.state.showVideoMenu ? 'toggled button-active' : ''
+            ]
+        };
+
+        return (
+            <AKInlineDialog
+                content = { <VideoQualityDialog /> }
+                isOpen = { _visible && this.state.showVideoMenu }
+                onClose = { this._onMenuClose }
+                position = { TOOLTIP_TO_DIALOG_POSITION[tooltipPosition] }>
```

Comment:
I see it passed in from SecondaryToolbar to make it show from the right and in PrimaryToolbar from the bottom (left if in filmstrip only). But it also looks like I have some display issues there I'll have to fix when in the PrimaryToolbar.